packages/graphql-yoga/__tests__/accept-header.spec.ts

```diff
@@ -166,7 +166,11 @@ describe('accept header', () => {
         },
       },
     )
-    expect(response.status).toEqual(406)
+    expect(response.status).toEqual(200)
+    const body = await response.json()
+    expect(body.errors).toEqual([
+      { message: 'Subscriptions are not supported' },
```

Comment:
Message is not clear for me. It is not supported because `Accept` doesn't conform the response's content type. `406` explains this better.

Reply:
Same as above, when accepting `application/json` the server should not use 4xx and 5xx ([see spec](https://graphql.github.io/graphql-over-http/draft/#sec-application-json)).

Reply:
Same with above, we don't accept `application/json` in this case, so we should return `Not Acceptable` to the client. Also I think error message is too generic. It is hard to understand that the problem is the accepted mime type.

Reply:
https://github.com/dotansimha/graphql-yoga/pull/1589#discussion_r945495511. I know that as per the HTTP spec 406 is obvious, but I am trying to follow the GQL over HTTP spec... I can improve the error message, it is indeed vague.